### PR TITLE
use modulo === 0 instead of implicit divisibility in fizzbuzz solution

### LINKED ISSUE #27: Fizz-buzz test solutions are misleading
https://github.com/rmurphey/js-assessment/pull/95 demonstrates out a potential area of confusion in [the fizzbuzz solutions](https://github.com/rmurphey/js-assessment-answers/blob/master/app/flowControl.js#L21-L29), which #26 attempts to address.

The issue is that the modulo operator returns 0 when it's *not divisible*, which evaluates to false in the if statement. The solutions, therefore, test the opposite of what one would expect.

To describe it another way, when `num % 5` is *true*, the expression evaluates to *false*.

#26 does a good job at trying to clarify this with words, but @ashleygwilliams proposed a code change that makes the code itself more expressive.

The idea would just be to do something like:

```js
      // divisible by 3 but not 5
      if (num % 3 === 0 && num % 5 !== 0) {
        return 'fizz';
      }

      // divisible by 5 but not 3
      if (num % 5 === 0 && num % 3 !== 0) {
        return 'buzz';
      }
```

It may even be simpler by just having one expression in each `if`. I'm not sure, 'cuz I don't have the tests open. But, you know, the idea is to use `===` to make the code more explicit.

Either this solution or #26 would close this issue. At the moment, I can't say which proposal I prefer!

### PR DESCRIPTION
**Pull request by ashleygwilliams**
fixes #27, review @jmeas?

### CODE REVIEW

File: app/flowControl.js [hunk 1]
```diff
@@ -11,25 +13,35 @@ define(function() {
...
-      if (num % 3 && num % 5) {
-        return num;
+      // make sure the value provided was a number, if not, return false
+      if (typeof num !== 'number') {
+        return false;
       }
 
-      // divisible by 3 but not 5
-      if (num % 5) {
+      // if the number is divisble by 3 AND 5, then when divided by both,
```
Comment (jmeas) [inline app/flowControl.js]: divisible*

**Comment by jmeas** [comment]
lgtm other than the comment above!

**Comment by ashleygwilliams** [comment]
another sanity check @jmeas ?

File: app/flowControl.js [hunk 2]
```diff
@@ -11,25 +13,35 @@ define(function() {
...
-        return num;
+      // make sure the value provided was a number, if not, return false
+      if (typeof num !== 'number') {
+        return false;
       }
 
-      // divisible by 3 but not 5
-      if (num % 5) {
+      // if the number is divisible by 3 AND 5, then when divided by both,
+      // the remainder will be zero, respectively
```
Comment (kadamwhite) [inline app/flowControl.js]: "remainder for each operation will be zero" feels less confusing than "remainder will be zero, respectively"

**Comment by kadamwhite** [comment]
One comment on wording otherwise :+1:

File: app/flowControl.js (hunk 2 above)
Comment (jmeas) [inline app/flowControl.js]: Good catch @kadamwhite!

Comment (ashleygwilliams) [inline app/flowControl.js]: hhaha ok on it

**Pull request merged**